### ISSUE TITLE
tests/data_files/Makefile throws warnings in 2.16

### ISSUE THREAD

**Issue by mpg** [ARMmbed/mbedtls#2591]
### Description
- Type: Bug 
- Priority: Minor

## Bug

**OS**  
Linux

**mbed TLS build:**  
Version: 2.16 branch (does not seem to affect development branch, did not try 2.7)

**Steps to reproduce**  

```
% cd tests/data_files && make
Makefile:805: warning: overriding recipe for target 'server1.req.cert_type'
Makefile:753: warning: ignoring old recipe for target 'server1.req.cert_type'
Makefile:809: warning: overriding recipe for target 'server1.req.key_usage'
Makefile:757: warning: ignoring old recipe for target 'server1.req.key_usage'
Makefile:813: warning: overriding recipe for target 'server1.req.ku-ct'
Makefile:761: warning: ignoring old recipe for target 'server1.req.ku-ct'
Makefile:826: warning: overriding recipe for target 'server5.req.ku.sha1'
Makefile:745: warning: ignoring old recipe for target 'server5.req.ku.sha1'
make: Nothing to be done for 'default'.
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-2842

**Comment by gilles-peskine-arm** [comment]
This has since been fixed in 2.16. 2.7 was also affected and only partially fixed, I fixed a remaining issue in https://github.com/ARMmbed/mbedtls/pull/3597.

I'm closing this issue now because it won't be closed automatically (since development is not affected), and technically 2.16 is already fixed, and there's a pending PR to fix 2.7, and I don't think this warrants a non-regression test (we don't even exercise this makefile on the CI at all).
